packages/mdc-checkbox/_mixins.scss

```diff
@@ -30,47 +33,202 @@
 // Public
 //
 
+@mixin mdc-checkbox($query: mdc-feature-all()) {
+  $feat-animation: mdc-feature-create-target($query, animation);
+  $feat-color: mdc-feature-create-target($query, color);
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-ripple-common($query);
+
+  @include mdc-feature-targets($feat-animation) {
+    @include mdc-checkbox-mark-keyframes_;
+  }
+
+  .mdc-checkbox {
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-checkbox-base_;
+    }
+
+    @include mdc-ripple-surface($query);
+    @include mdc-states($mdc-checkbox-baseline-theme-color, $query: $query);
+
+    @include mdc-feature-targets($feat-structure) {
+      @include mdc-ripple-radius-unbounded;
+    }
+  }
+
+  @include mdc-feature-targets($feat-color) {
+    @include mdc-checkbox-ink-color($mdc-checkbox-mark-color);
+    @include mdc-checkbox-focus-indicator-color($mdc-checkbox-baseline-theme-color);
+  }
+
+  @include mdc-checkbox-container-colors($query: $query);
```

Comment:
This looks safe to move up so the two surrounding color blocks can be combined. (I'd opt for up rather than down so disabled still comes later.)